### PR TITLE
Delete logfiles generated by tinytex

### LINKED ISSUE #260: Mechanism to ignore warnings or remove log files?
When the table is large and the output is "pdf" there is a warning because the table is truncated. As shown in #234, this can be solved with `theme_tt("resize")` but the fact that there's a warning also means that there's a `.log` file in the output directory. When several tables are generated like this, it creates a bunch of useless files. 

``` r
library(tinytable)

# wide df
test <- cbind(mtcars[1, ], mtcars[1, ])

tt(test) |> 
  theme_tt("resize") |> 
  save_tt("foo.pdf")
#> Warning: Package tabularray Warning: Table width is too small, need 284.97998pt
#> more!
```

``` r

list.files(getwd())
#> [1] "foo.pdf"                    "hurt-puma_reprex.R"        
#> [3] "hurt-puma_reprex.spin.R"    "hurt-puma_reprex.spin.Rmd" 
#> [5] "ideyec26lr3vl8l7hv62yz.log"  #<<<<<<<<<<<<<<<<<<<< this one
```

I know this is quite low-priority, but it would nice to have a mechanism to automatically clean up the output directory at some point.

### PR DESCRIPTION
**Pull request by etiennebacher**
Close #260 

I started with a basic implementation, let me know if you want to make this behavior opt-in.

### CODE REVIEW

**Comment by vincentarelbundock** [comment]
Let me know what you think. I'm happy to merge when you're ready.

File: R/save_tt.R [hunk 1]
```diff
@@ -51,24 +51,24 @@ save_tt <- function(x, output, overwrite = FALSE) {
...
+    "html" = "html",
+    "pdf" = "latex",
+    "tex" = "latex",
+    "md" = "markdown",
+    "Rmd" = "markdown",
+    "qmd" = "markdown",
+    "txt" = "markdown",
+    "docx" = "markdown",
+    "typ" = "typst",
+    stop("The supported file extensions are: .png, .html, .pdf, .tex, and .md.", call. = FALSE)
```
Comment (etiennebacher) [inline R/save_tt.R:64]: I think you forgot some extensions in the error message compared to choices in `switch()`

File: R/save_tt.R [hunk 2]
```diff
@@ -102,21 +101,35 @@ save_tt <- function(x, output, overwrite = FALSE) {
...
+    # clean
+    flag <- getOption("tinytable_save_pdf_clean", default = TRUE)
+    assert_flag(flag, name = "tinytable_save_pdf_clean")
+    if (flag) {
+      new_log_files <- setdiff(
+        list.files(temp_wd, pattern = "\\.log$", full.names = TRUE),
+        existing_log_files
+      )
+      invisible(file.remove(new_log_files))
+      unlink(f)
```
Comment (etiennebacher) [inline R/save_tt.R]: shouldn't this line be outside this `if` condition?

**Review by etiennebacher — COMMENTED** [review]
Thanks for polishing this, LGTM, just a couple of comments

File: R/save_tt.R [hunk 3]
```diff
@@ -90,23 +89,47 @@ save_tt <- function(x, output, overwrite = FALSE) {
...
+    # render
+    engine <- getOption("tinytable_save_pdf_engine", default = "xelatex")
+    assert_choice(engine, c("xelatex", "pdflatex", "lualatex"), name = "tinytable_save_pdf_engine")
+    if (engine == "xelatex") {
+      tinytex::xelatex(f, pdf_file = output)
+    } else if (engine == "pdflatex") {
+      tinytex::pdflatex(f, pdf_file = output)
+    } else if (engine == "lualatex") {
+      tinytex::lualatex(f, pdf_file = output)
+    }
```
Comment (etiennebacher) [inline R/save_tt.R:114]: You could simplify like this: `latexmk(f, pdf_file = output, engine = engine")`

**Comment by etiennebacher** [comment]
Also, is this worth a NEWS bullet?

**Comment by vincentarelbundock** [comment]
Thanks for the code and comments. This is a very nice addition. Merged.

**Pull request merged**